src/hooks/useCurrentUser.ts

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { fetchCurrentUser } from "@/api/currentUserApi";
+
+export function useCurrentUser() {
```

Comment:
This is a good place to cache the user. You can use `localStorage` directly or use one of wrappers around caching APIs, e.g. https://www.npmjs.com/package/localstorage-slim or `localforage`